src/postings/segment_postings.rs

```diff
@@ -98,6 +103,81 @@ impl<'a> DocSet for SegmentPostings<'a> {
         }
     }
 
+    fn skip_next(&mut self, target: DocId) -> SkipResult {
+        if !self.advance() {
+            return SkipResult::End;
+        }
+
+        let mut pos = self.index_within_block();
+        // skip blocks until one that might contain the target
+        loop {
+            // check if we need to go to the next block
+            if target > self.block_decoder.output(self.block_len - 1) {
+                self.cur += Wrapping(self.block_len - pos);
+                self.load_next_block();
+                pos = 0;
+
+                // there was no more data
+                if self.block_len == 0 {
+                    return SkipResult::End;
+                }
+            } else if target < self.block_decoder.output(pos) {
+                // We've overpassed the target after the first `advance` call
+                // or we're at the beginning of a block.
+                // Either way, we're on the first `DocId` greater than `target`
+                return SkipResult::OverStep;
+            } else {
+                break;
+            }
+        }
+
+        assert!(target >= self.block_decoder.output(pos));
+        assert!(target <= self.block_decoder.output(self.block_len - 1));
+
+        // we're in the right block, do a binary search
+        let mut start = pos;
+        let mut count = self.block_len - start;
+        while count > 0 {
+            let step = count / 2;
+            let mid = start + step;
+            let doc = self.block_decoder.output(mid);
+            if doc < target {
+                start = mid + 1;
+                count -= step + 1;
```

Comment:
maybe add a `debug_assert!` with the invariant `count == self.block_len - start`